### ISSUE TITLE
Port to gtksourceview-5 for async completion operations and hovers

### ISSUE THREAD

**Issue by esodan** [elementary/code#1066]
## Problem
Code could provider better support for completion and gain a hover using gtksourceview-5, for these reasons:

1. Completions should be async in order to help run async operations to calculate items, like when using LSP environments where a client request to the server completion items asynchronously, new API in gtksourceview-5 performs populate items async.
2. New gtksourceview-5 has support to provide hovers, that is an important features for code editors in order to show context information like LSP environments requesting information to servers on the current context, in version 5 populate hover is also async and is required for LSP too in order to avoid UI block.

## Proposal
Port Code to use gtksourceview-5 in order to increase its compatilibility with LSP servers.

**Comment by esodan** [comment]
Big issue is gtksourceview-5 requires glib-2.0 >= 66 and gtk-4, because elementary OS will not be ported to gtk 4.0 any time soon, so this issue should wait.

**Comment by kaixoo** [comment]
Is someone working in this right now?

**Comment by jeremypw** [comment]
No-one is working on the gtk4 port right now.  There are a lot of unmerged gtk3 PRs.
